### PR TITLE
demisto/etc#23067 Improving unclear error messages

### PR DESCRIPTION
**Pull request by hod-alpert**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/23067

## Description
- Printing fix suggestions with json.dumps or yaml.parse according to the file
- ignoring _unified files
- Making release notes error message more readable
## Screenshots
Paste here any images that will help the reviewer

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: demisto_sdk/commands/common/hook_validations/base_validator.py [hunk 1]
```diff
@@ -67,7 +67,7 @@ def is_there_release_notes(self):
             # check release_notes file exists and contain text
             if release_notes is None:
                 self.is_valid = False
-                print_error("Missing release notes for: {}".format(self.file_path))
+                print_error("Missing release notes for: {} in {}".format(self.file_path, rn_path))
```
Comment (Itay4) [inline demisto_sdk/commands/common/hook_validations/base_validator.py:70]:
```suggestion
                print_error(f"Missing release notes for: {self.file_path}\n Create or update the file {rn_path} with the relevant changes.")
```

File: demisto_sdk/commands/common/hook_validations/structure.py [hunk 2]
```diff
@@ -326,3 +333,30 @@ def parse_error_line(self, err) -> str:
...
                                                               ))
+
+    @staticmethod
+    def get_content_parser(file_path: str) -> Callable[[str], str]:
+        """
+        Return a method with which 'curr' (the current key the lies in the path of the error) should be printed with
+        If the file is a yml file:
+            will return a yaml.dump function
+        If the file is a json file:
+            will return a json.dupms function configured with indent=4
```
Comment (Itay4) [inline demisto_sdk/commands/common/hook_validations/structure.py:344]:
```suggestion
            will return a json.dumps function configured with indent=4
```

File: demisto_sdk/commands/common/hook_validations/structure.py [hunk 3]
```diff
@@ -326,3 +333,30 @@ def parse_error_line(self, err) -> str:
             return (
                 'Failed: {} failed.\nMissing {} in {}'.format(self.file_path, str(key_from_error), "root",
                                                               ))
+
+    @staticmethod
+    def get_content_parser(file_path: str) -> Callable[[str], str]:
```
Comment (Itay4) [inline demisto_sdk/commands/common/hook_validations/structure.py:338]: feels to me this should be in tools, what do you think?
also the func name is not very indicative, what about something like `get_content_file_type_dump` ?

File: demisto_sdk/commands/validate/file_validator.py [hunk 4]
```diff
@@ -276,13 +276,15 @@ def validate_modified_files(self, modified_files):  # noqa: C901
...
             # modified_files are returning from running git diff.
             # If modified file was renamed\moved, file_path could be a tuple containing original path and new path
-            file_type = find_type(file_path[1]) if isinstance(file_path, tuple) else find_type(file_path)
-            old_file_path = None
             if isinstance(file_path, tuple):
                 old_file_path, file_path = file_path
-
+            file_type = find_type(file_path)
+            # unified files should not be validated
+            if '_unified' in file_path:
```
Comment (Itay4) [inline demisto_sdk/commands/validate/file_validator.py:286]: lets check `.endswith("_unified.yml")`

File: demisto_sdk/commands/common/tests/structure_test.py [hunk 5]
```diff
@@ -206,18 +207,18 @@ def test_print_error_msg(self, path, scheme, error, correct, mocker):
     ]
 
     @pytest.mark.parametrize('file_path, expected', CONTENT_PARSER_INPUT)
-    def test_get_content_parser(self, file_path: str, expected: Callable[[str], str]):
+    def test_get_content_file_type_dump(self, file_path: str, expected: Callable[[str], str]):
```
Comment (Itay4) [inline demisto_sdk/commands/common/tests/structure_test.py:210]: lets move test to tools test

**Comment by Itay4** [comment]
@hod-alpert please note my comment about testing the unified files

**Comment by hod-alpert** [comment]
@Itay4 Added tests

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,6 +1,9 @@
 # Changelog
 * Fixed a bug in the **unify** command when output path was provided empty.
 * Improved error message for integration with no tests configured
+* Missing release notes message will now show the release notes file path to update.
+* unified yml files will be ignored.
```
Comment (Itay4) [inline CHANGELOG.md:5]:
```suggestion
* Fixed an issue in the **validate** command in which unified YAML files were not ignored.
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,6 +1,9 @@
...
+* Fix suggestions will be in the relevant file format.
```
Comment (Itay4) [inline CHANGELOG.md:6]:
```suggestion
* File format suggestions are now shown in the relevant file format (JSON or YAML).
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,6 +1,9 @@
 # Changelog
 * Fixed a bug in the **unify** command when output path was provided empty.
 * Improved error message for integration with no tests configured
+* Missing release notes message will now show the release notes file path to update.
```
Comment (Itay4) [inline CHANGELOG.md:4]:
```suggestion
* Missing release notes message are now showing the release notes file path to update.
```

**Pull request merged**